scripts/shellcheck.sh

```diff
@@ -0,0 +1,322 @@
+
+example: $ ./scripts/shellcheck.sh -C auto -S info -e SC2034 -e SC2086
+
+EOF
+}
+
+parse_cli_args() {
+    echo
+    while [[ $# -gt 0 ]]; do
+        optarg=
+        shft_cnt=1
+        # if -- is passed then stop parsing
+        if [[ "$1" = '--' ]]; then
+            break
+        # --option=value
+        elif [[ "$1" =~ ^-- && ! "$1" =~ ^--$ ]]; then
+            optarg="${1#*=}"; shft_cnt=1
+        # -o=value
+        elif [[ "$1" =~ ^-- && $# -gt 1 && ! "$2" =~ ^- ]]; then
+            optarg="$2"; shft_cnt=2
+        # -o value
+        elif [[ "$1" =~ ^-[^-] && $# -gt 1 && ! "$2" =~ ^- ]]; then
+            optarg="$2"; shft_cnt=2
+        # -o
+        elif [[ "$1" =~ ^-[^-] ]]; then
+            optarg="${1/??/}"
+        fi
+
+        case "$1" in
+            -S*|--severity*)
+                case "${optarg}" in
+                    style|info|warning|error)
+                        SEVERITY="${optarg}" ;;
+                    *)
+                        exit_bad_arg "$optarg" ;;
+                esac
+                shift "$shft_cnt" ;;
+            -C*|--color*)
+                case "${optarg}" in
+                    always|auto|never)
+                        COLOR="${optarg}" ;;
+                    *)
+                        exit_bad_arg "$optarg" ;;
+                esac
+                shift "$shft_cnt" ;;
+            -e*|--exclude-code*)
+                # valid if matching format SC1000-SC9999
+                if [[ "${optarg}" =~ ^SC[0-9]{4}$ ]]; then
+                    # if empty, fill with error code, else append comma and new code
+                    if [[ -z "${EXCLUDE_CODES}" ]]; then
+                        EXCLUDE_CODES+="${optarg}"
+                    else
+                        EXCLUDE_CODES+=",${optarg}"
+                    fi
+                else
+                    exit_bad_arg "$optarg"
+                fi
+                shift "$shft_cnt" ;;
+            -v|--view)
+                VIEW="True"
+                shift "$shft_cnt" ;;
+            -h|--help)
+                echo; echo "Command line arguments: $0 $*"; echo
+                show_usage
+                exit 0 ;;
+            *)
+                exit_bad_arg "$optarg" ;;
+        esac
+    done
+}
+
+get_num_cores() {
+    local CORE_COUNT=1
+    if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+        CORE_COUNT=$(grep -c ^processor /proc/cpuinfo)
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        CORE_COUNT=$(sysctl -n hw.ncpu)
+    fi
+    printf "%d\n" "$CORE_COUNT"
+}
+
+run_shellcheck() {
+    # check if git is installed
+    if command -v git &>/dev/null; then
+        if [[ -z "$EXCLUDE_CODES" ]]; then
+            git ls-files '*.sh' | xargs -n 1 -P "$NUM_CORES" shellcheck -s bash -C"$COLOR" > "$SHELLCHECK_REPORT_ALL_MSGS"
+            git ls-files '*.sh' | xargs -n 1 -P "$NUM_CORES" shellcheck -s bash -C"$COLOR" -S "$SEVERITY" > "$SHELLCHECK_REPORT"
+        else
```

Comment:
can just pass empty or populated exclude codes into args and it will work, so can get rid of extra if/else